src/regfunc.jl

```diff
@@ -0,0 +1,402 @@
+#-----------------------------------------------------------
+#-----------------------------------------------------------
+# Define function to access windows registry
+# -- part of WinReg Module --
+#----------------------------------------------------------
+#----------------------------------------------------------
+
+
+"""
+   regread(key::AbstractString, valuename::AbstractString)
+
+Read a value with `valuename` into `key`. 
+`key` can be specified using abbreviated form (e.g. 'HKEY_LOCAL_MACHINE' -> 'HKLM').
+Return type in Julia is based onto value type stored in registry; returns `nothing` (type is `Void`) if key not found"""
+function regread(key::AbstractString, valuename::AbstractString)
+	root = getroothkey(key)
+	if root == 0
+		return nothing
+	end
+	subkey = getsubkey(key)
+	hk::HKEY = 0
+	len = Ref{DWORD}(0)
+	typ = Ref{DWORD}(0)
+	
+	buf = nothing
+	try
+		hk = regopenkeyex(root, subkey, REG_SAM.KEY_READ)
+		#get buffer dim
+		res = ccall((:RegQueryValueExW, advapi), stdcall,
+		  LONG,
+		  (HKEY, Cwstring, LPDWORD, LPDWORD, LPBYTE, LPDWORD),
+		  hk, valuename, C_NULL, typ, C_NULL, len )
+		if res == ERROR_SUCCESS
+			buf = Vector{UInt8}(len[])
+			len = Ref{DWORD}(length(buf))
+			
+			res = ccall((:RegQueryValueExW, advapi), stdcall,
+			  LONG,
+			  (HKEY, Cwstring, LPDWORD, LPDWORD, LPBYTE, LPDWORD),
+			  hk, valuename, C_NULL, typ, pointer(buf), len )
+			
+			regclosekey(hk)
+			
+			if res != ERROR_SUCCESS
+				return nothing
+			end
+		else
+			regclosekey(hk)
+			
+			error("error getting value length")
+		end
+	finally
+		regclosekey(hk)
+	end
+	if hk == 0
+		return nothing
+	end
+	
+	if typ[] == REG_TYPE.REG_SZ || typ[] == REG_TYPE.REG_EXPAND_SZ || typ[] == REG_TYPE.REG_LINK
+		wbuf = reinterpret(Cwchar_t, buf)
+		l = len[]
```

Comment:
Not at all: I've not covered this case because if queryvalue() is not successful, then error is returned (line 50); instead it cover case speficied in 'Remarks' paragraph of MS documentation.
I believe your case is not applicable, because at 1st call I pass NULL data, never returning ERROR_MORE_DATA.